libpympb/pympb.cpp

```diff
@@ -809,7 +797,17 @@ void mode_solver::init(int p, bool reset_fields, geometric_object_list geometry,
   evectmatrix_flops = eigensolver_flops;
 }
 
-void mode_solver::init_epsilon(geometric_object_list *geometry) {
+void mode_solver::init_epsilon(geometric_object_list *geometry_in) {
+  // Persist geometry data and move it out of the input argument.
+  clear_geometry_list();
+  if (geometry_in->num_items && geometry_in->items) {
+    geometry_list.items = new geometric_object[geometry_in->num_items];
+    std::copy_n(geometry_in->items, geometry_in->num_items, geometry_list.items);
+    geometry_list.num_items = geometry_in->num_items;
+    delete [] geometry_in->items;
+    geometry_in->items = NULL;
```

Comment:
Why not just set `geometry_list.items = geometry_in->items`?

Reply:
Done.